### PR TITLE
Transform lease TTLs into ints

### PR DESCRIPTION
**Pull request by akerouanton**
The `default_lease_ttl` and `max_lease_ttl` parameters in
hashivault_secret_engine can take either an integer or a string with a
duration suffix. This mimics what Vault API's `PUT /sys/mounts/:path`
accepts. However, `GET /sys/mounts/:path` always return an integer (ie.
string durations are converted into an integer).

Because of that, hashivault_secret_engine module isn't idempotent when
string durations or string containing only an integer is passed to it.
Note that the latter case happens when this value comes from a variable
(eg. a dict representing multiple engines to mount).

This fix converts aforementioned parameters into integer before
comparing them with actual values. It also changes the doc to state
only integers are accepted and why so.

Signed-off-by: Albin Kerouanton <albinker@gmail.com>

### CODE REVIEW

File: ansible/modules/hashivault/hashivault_secret_engine.py [hunk 1]
```diff
@@ -81,6 +120,10 @@ def hashivault_secret_engine(module):
     backend = params.get('backend')
     description = params.get('description')
     config = params.get('config')
+    if 'default_lease_ttl' in config:
+        config['default_lease_ttl'] = int(config['default_lease_ttl'])
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_secret_engine.py:124]: My concern here is this looks like it isn't going to be backward compatible. Anyone with a default_lease_ttl that isn't an int now is going to fail, right?

Comment (akerouanton) [reply ansible/modules/hashivault/hashivault_secret_engine.py:124]: Yeah right, it's definetively a BC break. Although I forgot to mention it, I wanted to have your opinion on this. I'll add a function to parse these strings properly to not break backward compatibility :wink:

Comment (akerouanton) [reply ansible/modules/hashivault/hashivault_secret_engine.py]: @TerryHowe I fixed it :wink:

File: ansible/modules/hashivault/hashivault_secret_engine.py [hunk 2]
```diff
@@ -73,6 +112,24 @@ def main():
         module.exit_json(**result)
 
 
+def parse_duration(duration):
+    regex = re.compile(r'((?P<days>\d+)d)|((?P<hours>\d+)h)|((?P<minutes>\d+)m)|((?P<seconds>\d+)s)')
```
Comment (TerryHowe) [inline ansible/modules/hashivault/hashivault_secret_engine.py:116]: re wasn't imported, so the tests failed, but maybe just `if duration.endswith('d')` something like that?

**Comment by TerryHowe** [comment]
Thanks!

**Pull request merged**